zebra/zebra_rib.c

```diff
@@ -1316,20 +1316,23 @@ static bool rib_update_re_from_ctx(struct route_entry *re,
 	 * status.
 	 */
 
+	matched = false;
+	ctxnhg = dplane_ctx_get_ng(ctx);
+
 	/* Check both fib group and notif group for equivalence.
 	 *
 	 * Let's assume the nexthops are ordered here to save time.
 	 */
-	if (nexthop_group_equal(&re->fib_ng, dplane_ctx_get_ng(ctx)) == false) {
+	if (re->fib_ng.nexthop &&
+	    ctxnhg->nexthop &&
+	    nexthop_group_equal(&re->fib_ng, ctxnhg) == false) {
 		if (IS_ZEBRA_DEBUG_RIB_DETAILED) {
-			zlog_debug(
-				"%u:%s update_from_ctx: notif nh and fib nh mismatch",
-				re->vrf_id, dest_str);
+			zlog_debug("%u:%s update_from_ctx: notif nh and fib nh mismatch",
+				   re->vrf_id, dest_str);
 		}
 
 		matched = false;
-	} else
-		matched = true;
+	}
 
 	/* If the new FIB set matches the existing FIB set, we're done. */
 	if (matched) {
```

Comment:
matched is always false until here. So the if condition is a dead code.

Reply:
Quite right, thanks; I've pushed an update.